### ISSUE TITLE
I am wondering

### ISSUE THREAD

**Issue by Kek5chen** [rxn-ro/RHA#1]
I am wondering why this is a thing? Just generating a UID is much simpler and requires less processing power than this piece of code. Having this in production code would be a huge waste of processing power. I also do not get why it is called RHA? You are not reversing any hashing algorithms. The name itself does not make sense and is missleading because a hash is made to be irreversable.

It is also not a single header library because you are using the md5 files too. A single header library is nothing more than a single header file.

I am just wondering on what is going on here.

**Comment by rxn-ro** [comment]
My man, first of all the md5 file is just so the RHA hash is longer. Second of all, doesnt require that much processing power, if the software uses multi-threading everything is allright. And third of all , is called RHA because the final hash is reversed (backwords) so wont get cracked.

**Comment by Kek5chen** [comment]
1. even if it does use the file to create a longer hash, it doesnt matter, its still not a single header lib. single heder libs can be included by copying one, singular header file without any additional files.
2. even if its not much processing power, it is still more than needed. in big applications you try to be as resource efficient as possible.
3. if your lib is open source, and the name even suggests that the final string is reversed, it doesnt help with crack protection.
4. you cant crack md5 because it drops bytes from the original input and therefore makes it unreversable.


so basically your program is just a waste of processing power in the end.

**Comment by Kek5chen** [comment]
also, in your readme it says:

`The algorithm does not have a decrypt function, being a 1-way hashing function.`: it is not possible to decrypt a hash, and every hashing function is one-way, because, like i said earlier, you drop bytes.

therefore..

`RHA::rha_encrypt` also does not make sense in any way. also, maybe consider not namespacing the function if you define the function name to include the namespace name anyways.

`encrypting data`: you are not encrypting anything, you are encoding, please look up the difference and learn about it before making any statements.

I am trying to help you out and give you some tips and important informtion on your programming journey, it is sad that you talk to me so disrespectfully. I get that you feel offended since you probably havent been coding and hacking for that long, but please understand that people are mostly just trying to help you out.

If you are looking to stay in a toxic and non productive environment though, keep cracking and hacking, maybe join some hacking discord servers where people say the n word more than they send messages.

**Comment by rxn-ro** [comment]
> 1. Its not about dragging the files. Its about you saying its a single header file when its not.
> 2. Its not about that there is not enough processing power, but why would people waste ram if its not needed to.
> 3. hashes are supposed to be very random by default. if you just md5 hash rand() there will be something completely different and unpredictable every time. if you just generate a random 25 char long string, that will be just as random for less processing power.
> 4. you. cant. crack. hashes.
> 5. please go learn about md5 and hash algorithms first before you start a project. i get that it is a nice training to learn c++, but thats why i asked what this is for. if this should be an actual project i recommend anyone that finds this to not use it and to just generate a random string.
> 
> it is a waste of processing power, because even though multithreading might help, theres no need to waste the power in the first place since its just unnecessary. 

u again dude? tell me what experience u got in the domain, then talk, thanks.
